src/utils/tekton-results.ts

```diff
@@ -167,8 +183,6 @@ export const selectorToFilter = (selector?: Selector) => {
       if (matchExpressions) {
         filter = AND(filter, expressionsToFilter(matchExpressions));
       }
-    } else {
-      filter = labelsToFilter(selector as MatchLabels);
```

Comment:
This is overwriting the filter. If a selector expression doesn't include a matchlabel or matchexpressions it will always execute the else part which is setting the incorrect filter hence the change in test output as well.  `filterByName` should have a `startsWith` expression instead of a label key assertion.